src/NetscriptFunctions/Singularity.ts

```diff
@@ -555,6 +555,22 @@ export function NetscriptSingularity(
         return true;
       }
 
+      const programsRef = player.getHomeComputer().programs;
+      // Remove partially created program if there is one
+      const existingPartialExeIndex = programsRef.findIndex(
+        (program) => item?.program && program.startsWith(item?.program),
+      );
+      // findIndex returns -1 if there is no match, we only want to splice on a match
+      if (existingPartialExeIndex > -1) {
+        programsRef.splice(existingPartialExeIndex, 1);
```

Comment:
hmm would removing a partial program cause issues if they are working on it? maybe put the cancel work above juust incase